### PR TITLE
[USM] Create benchmark for consumer operations

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?

Creates a benchmark function for the consumer operations.

### Motivation

To address the current user-mode bottlenecks, this benchmark serves as a foundational diagnostic tool to determine whether the improvements are effective and have a impact on the current process.

### CODE REVIEW

File: pkg/network/protocols/events/consumer_test.go [hunk 1]
```diff
@@ -197,3 +201,38 @@ func TestInvalidBatchCountMetric(t *testing.T) {
 
 	require.Equalf(t, int(consumer.invalidBatchCount.Get()), 1, "invalidBatchCount should be greater than 0")
 }
+
```
Comment (yuri-lipnesh) [inline pkg/network/protocols/events/consumer_test.go:204]: Documentation on the purpose of this benchmark would be helpful.

File: pkg/network/protocols/events/consumer_test.go [hunk 2]
```diff
@@ -197,3 +201,38 @@ func TestInvalidBatchCountMetric(t *testing.T) {
...
+func BenchmarkConsumer(b *testing.B) {
+	kversion, err := kernel.HostVersion()
+	require.NoError(b, err)
+	if minVersion := kernel.VersionCode(4, 14, 0); kversion < minVersion {
+		b.Skipf("package not supported by kernels < %s", minVersion)
```
Comment (yuri-lipnesh) [inline pkg/network/protocols/events/consumer_test.go:209]: maybe to print current kernel version as well?

File: pkg/network/protocols/events/consumer_test.go [hunk 3]
```diff
@@ -197,3 +201,38 @@ func TestInvalidBatchCountMetric(t *testing.T) {
...
+	}
+
+	const numEvents = 1000
+	c := config.New()
+	program, err := newEBPFProgram(c)
+	require.NoError(b, err)
+
```
Comment (yuri-lipnesh) [inline pkg/network/protocols/events/consumer_test.go:216]: need b.Cleanup() to stop/clean eBPF program

File: pkg/network/protocols/events/consumer_test.go [hunk 4]
```diff
@@ -197,3 +201,38 @@ func TestInvalidBatchCountMetric(t *testing.T) {
...
 }
+
+func BenchmarkConsumer(b *testing.B) {
+	kversion, err := kernel.HostVersion()
+	require.NoError(b, err)
+	if minVersion := kernel.VersionCode(4, 14, 0); kversion < minVersion {
+		b.Skipf("package not supported by kernels < %s", minVersion)
+	}
+
+	const numEvents = 1000
```
Comment (yuri-lipnesh) [inline pkg/network/protocols/events/consumer_test.go:212]: how do you know this number of events is good enough?

File: pkg/network/protocols/events/consumer_test.go [hunk 5]
```diff
@@ -197,3 +201,38 @@ func TestInvalidBatchCountMetric(t *testing.T) {
...
+	consumer, err := NewConsumer("test", program, func(_ []uint64) {})
+	require.NoError(b, err)
+	generator := newEventGenerator(program, b)
+	require.NoError(b, program.Start())
+
+	for j := 0; j < numEvents; j++ {
+		require.NoError(b, generator.Generate(uint64(j)))
+	}
+	generator.Stop()
+	time.Sleep(100 * time.Millisecond)
```
Comment (yuri-lipnesh) [inline pkg/network/protocols/events/consumer_test.go:226]: what is purpose of this sleep? do you have any indicatior on how long to wait, not just an arbitrary number?

**Comment by amitslavin** [comment]
@yuri-lipnesh, thank you for the review, but this branch was a draft and not intended for review.

**Comment by yuri-lipnesh** [comment]
@amitslavin , Oh, I clicked on the link in the email notification, didn't notice it was a draft PR.

**Comment by amitslavin** [comment]
draft

**Pull request closed without merging**